src/wix/WixToolset.Core.Burn/Bundles/BurnReader.cs

```diff
@@ -142,6 +142,69 @@ public bool ExtractUXContainer(string outputDirectory, string tempDirectory)
             return true;
         }
 
+        /// <summary>
+        /// Extracts detached containers to the output directory.
+        /// </summary>
+        /// <param name="outputDirectory">Directory to write extracted files to.</param>
+        /// <param name="tempDirectory">Scratch directory.</param>
+        /// <param name="uxOutputDirectory">UX extraction folder. If null or empty, a UX folder will be created within tempDirectory</param>
+        /// <returns>True if successful, false otherwise</returns>
+        public bool ExtractDetachedContainers(string outputDirectory, string tempDirectory, string uxOutputDirectory)
+        {
+            if (string.IsNullOrEmpty(uxOutputDirectory))
+            {
+                uxOutputDirectory = Path.Combine(tempDirectory, "UX", "Final");
+            }
+
+            var manifestPath = Path.Combine(uxOutputDirectory, "manifest.xml");
+            if (!File.Exists(manifestPath))
+            {
+                var uxTempDirectory = Path.Combine(tempDirectory, "UX", "Temp");
+                if (!this.ExtractUXContainer(uxOutputDirectory, uxTempDirectory))
+                {
+                    return false;
+                }
+            }
+
+            Directory.CreateDirectory(tempDirectory);
+            Directory.CreateDirectory(outputDirectory);
```

Comment:
I don't think we need to create either of these directories here. They will be created below only if necessary.